src/main/java/seedu/address/logic/parser/ParserType.java

```diff
@@ -0,0 +1,38 @@
+package seedu.address.logic.parser;
+
+import java.lang.reflect.Field;
+import java.util.logging.Logger;
+
+import seedu.address.commons.core.LogsCenter;
+
+/*
+ * Iterates through all parsers to find the appropriate one for the given command input
+ */
+public class ParserType {
+		
+	private static final Class<?>[] parserTypes = CommandParserList.getList();
+    private static final Logger logger = LogsCenter.getLogger(ParserType.class);
+
+	
+	public static CommandParser get(String commandWord){
+		for(int i=0; i<parserTypes.length; i++){
+			try {
```

Comment:
Comments on exactly what this is doing will be useful because it isn't immediately obvious